src/bindings/c/include/openvino/c/ov_partial_shape.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a header file for partial shape C API, which is a C wrapper for ov::PartialShape class.
+ *
+ * @file ov_partial_shape.h
+ */
+
+#pragma once
+
+#include "openvino/c/ov_common.h"
+#include "openvino/c/ov_dimension.h"
+#include "openvino/c/ov_layout.h"
+#include "openvino/c/ov_rank.h"
+#include "openvino/c/ov_shape.h"
+
+typedef struct ov_partial_shape ov_partial_shape_t;
+
+// PartialShape
+/**
+ * @defgroup partial_shape partial_shape
+ * @ingroup openvino_c
+ * Set of functions representing PartialShape.
+ * @{
+ */
+
+/**
+ * @brief Create a partial shape and initialze with rank and dimension.
+ * @ingroup partial_shape
+ * @param rank support dynamic and static rank
+ * @param dims support dynamic and static dimension
+ *  Dynamic rank:
+ *     Example: "?"
+ *  Static rank, but dynamic dimensions on some or all axes.
+ *     Examples: "{1,2,?,4}" or "{?,?,?}" or "{1,2,-1,4}""
+ *  Static rank, and static dimensions on all axes.
+ *     Examples: "{1,2,3,4}" or "{6}" or "{}""
+ *
+ * @param ov_status_e a status code.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_partial_shape_create(ov_partial_shape_t** partial_shape_obj, ov_rank_t* rank, ov_dimensions_t* dims);
+
+/**
+ * @brief Parse the partial shape to readable string.
+ * @ingroup partial_shape
+ * @param ov_status_e a status code.
+ */
+OPENVINO_C_API(const char*) ov_partial_shape_to_string(ov_partial_shape_t* partial_shape);
+
+/**
+ * @brief Release partial shape.
+ * @ingroup partial_shape
+ * @param partial_shape will be released.
+ */
+OPENVINO_C_API(void) ov_partial_shape_free(ov_partial_shape_t* partial_shape);
+
+/**
+ * @brief Covert partial shape to static shape.
+ * @ingroup partial_shape
+ * @param ov_status_e a status code, return OK if successful
+ */
+OPENVINO_C_API(ov_status_e) ov_partial_shape_to_shape(ov_partial_shape_t* partial_shape, ov_shape_t* shape);
+
+/**
+ * @brief Covert shape to partial shape.
+ * @ingroup shape
+ * @param ov_status_e a status code, return OK if successful
+ */
+OPENVINO_C_API(ov_status_e) ov_shape_to_partial_shape(ov_shape_t* shape, ov_partial_shape_t** partial_shape);
+
```

Comment:
what is the purpose of having this API?
how can I extract dimensions from pshape and understand its rank? 

From current API I can only create, but cannot understand what is inside.

Can I create dynamic shape from just array of int pairs? E.g. I have { { 1, 2 }, { 2, 3 } } - rank is 2, dim1 = {1,2}, dim2 = {2,3}. Can we have simple API for this?

Reply:
As latest reviewer's comments, rank/dimension/shape/pshape all are defined as opaque handle, we have to provide additional api to make user understand it.
BTW, in C world, it will be not easy to define a simple dynamic size array to represent such array of int pairs as input parameter.
In original implement, we use a string to represent, but it was denied.

Reply:
In order to provide an opportunity to work with partial shape can we use 
```
typedef struct partial_shape {
    ov_rank* rank;
    ov_dimensions_t* dimensions;
} partial_shape_t;
```

Ilya said that at the current moment we don't have API to work and change partial shape. We cannot get partial shape from the model and after that change it.

Reply:
It makes sense if we adopt the new definition of ov_dimension and ov_rank, then partial_shape should be defined as:

```
typedef struct ov_partial_shape {
    ov_rank_t rank;
    ov_dimension_t *dims;
};
```
Here, we can provide clear C style structure for partial shape, the cost is that we need add additional conversion operations in C APIs - between C++ rank/dimension/pshape class and C definition.